cdap-proto/src/main/java/co/cask/cdap/proto/id/DatasetId.java

```diff
@@ -43,6 +44,10 @@ public DatasetId(String namespace, String dataset) {
     this.dataset = dataset;
   }
 
+  public DatasetId(MetadataEntity datasetEntity) {
```

Comment:
hmmmm so this assumes that the MetadaEntity is actually a dataset. If I construct a MetadataEntity.of(STREAM, "myStream"), then I can pass that in here, and it will create a dataset Id from it? Seems wrong... also, this does not deal with the case where the namespace is missing from the MetadataEntity. 

I guess this comment will apply to all other entity types, too. 

Perhaps it is better to 
1. have a static method in EntityId that inspects the tags in the metadata and creates the correct id?
2. or keep this method, but use EntityId.ensureIsValidDatasetId() to make sure we don't cast apples into oranges.

Reply:
I like the 1 idea more. Will switch to that in later prs. For now I will remove it since it does not affect the apis.